### PR TITLE
Add play_pause action support

### PR DESCRIPTION
**Pull request by maxi1134**
Add play_pause action support in order to support mini media player default action for media control (play_pause)

### CODE REVIEW

File: custom_components/media_player_template/media_player.py [hunk 1]
```diff
@@ -109,9 +110,10 @@
         vol.Optional(CONF_ENTITY_PICTURE_TEMPLATE): cv.template,
         vol.Optional(CONF_AVAILABILITY_TEMPLATE): cv.template,
         vol.Optional(CURRENT_SOURCE_TEMPLATE): cv.template,
-        vol.Optional(ON_ACTION): cv.SCRIPT_SCHEMA,
-        vol.Optional(OFF_ACTION): cv.SCRIPT_SCHEMA,
+        vol.Required(ON_ACTION): cv.SCRIPT_SCHEMA,
+        vol.Required(OFF_ACTION): cv.SCRIPT_SCHEMA,
```
Comment (Sennevds) [inline custom_components/media_player_template/media_player.py:114]: Why did you make them required?

Comment (maxi1134) [reply custom_components/media_player_template/media_player.py:114]: I'll be 100% honest. Python is not something I code regularly.
I mainly needed play_pause to be able to use mini_media_player card.

I simply added the same entries as play_action but adapted it for play_pause!

Comment (Sennevds) [reply custom_components/media_player_template/media_player.py:114]: Yeah the changes are okay but you've change the ON_ACTION/OFF_ACTION to required parameters

Comment (maxi1134) [reply custom_components/media_player_template/media_player.py:114]: I did not modify those lines AFAIK. I took the code on your page and added play_pause in 8 locations.

File: custom_components/media_player_template/manifest.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "requirements": [],
     "dependencies": [],
     "after_dependencies": ["template"],
-    "codeowners": ["@Sennevds"],
-    "version": "1.0.0",
+    "codeowners": ["@Maxi1134"],
```
Comment (golles) [inline custom_components/media_player_template/manifest.json:9]: Please revert this
```suggestion
    "codeowners": ["@Sennevds"],
```

File: custom_components/media_player_template/media_player.py [hunk 3]
```diff
@@ -109,9 +109,10 @@
...
```
Comment (golles) [inline custom_components/media_player_template/media_player.py:113]: Should indeed be reverted
```suggestion
        vol.Optional(ON_ACTION): cv.SCRIPT_SCHEMA,
        vol.Optional(OFF_ACTION): cv.SCRIPT_SCHEMA,
```

File: custom_components/media_player_template/media_player.py [hunk 4]
```diff
@@ -169,9 +171,10 @@ async def _async_create_entities(hass, config):
         entity_picture_template = device_config.get(CONF_ENTITY_PICTURE_TEMPLATE)
         availability_template = device_config.get(CONF_AVAILABILITY_TEMPLATE)
         current_source_template = device_config.get(CURRENT_SOURCE_TEMPLATE)
-        on_action = device_config.get(ON_ACTION)
-        off_action = device_config.get(OFF_ACTION)
+        on_action = device_config[ON_ACTION]
+        off_action = device_config[OFF_ACTION]
```
Comment (golles) [inline custom_components/media_player_template/media_player.py:175]: Please revert this
```suggestion
        on_action = device_config.get(ON_ACTION)
        off_action = device_config.get(OFF_ACTION)
```

File: custom_components/media_player_template/media_player.py [hunk 5]
```diff
@@ -308,19 +313,16 @@ def __init__(
...
-        self._on_script = None
-        if on_action is not None:
-            self._on_script = Script(hass, on_action, friendly_name, self._domain)
-
-        self._off_script = None
-        if off_action is not None:
-            self._off_script = Script(hass, off_action, friendly_name, self._domain)
-
+        self._on_script = Script(hass, on_action, friendly_name, self._domain)
+        self._off_script = Script(hass, off_action, friendly_name, self._domain)
```
Comment (golles) [inline custom_components/media_player_template/media_player.py:317]: I think this should also be reverted

File: custom_components/media_player_template/media_player.py [hunk 6]
```diff
@@ -519,11 +521,7 @@ def entity_picture(self):
     def supported_features(self):
         """Flag media player features that are supported."""
 
-        support = 0
-        if self._on_script is not None:
-            support |= SUPPORT_TURN_ON
-        if self._off_script is not None:
-            support |= SUPPORT_TURN_OFF
+        support = SUPPORT_TEMPLATE
```
Comment (golles) [inline custom_components/media_player_template/media_player.py:524]: And this should also be reverted

File: custom_components/media_player_template/media_player.py [hunk 7]
```diff
@@ -143,6 +144,7 @@
         vol.Optional(CURRENT_SOUND_MODE_TEMPLATE): cv.template,
     }
 )
+SUPPORT_TEMPLATE = SUPPORT_TURN_OFF | SUPPORT_TURN_ON
```
Comment (golles) [inline custom_components/media_player_template/media_player.py:147]: Revert this

**Comment by golles** [comment]
This PR does contain a lot of changes that shouldn't be part of this change and should be reverted

**Comment by maxi1134** [comment]
I did a fork for myself. I have no interest in modifying anything for a pull. My coding abilities are far from good.

**Pull request closed without merging**